### PR TITLE
Token reward harvesting

### PR DESCRIPTION
**Pull request by anonymoussprocket**
- added support for claiming Plenty rewards from farms
- added support for kDAO along with reward claims
- bug fixes for display of multi-token FA2 contracts, like the WRAP family
- updated Beacon dApp operation parsing
- updated contract addresses
- fixed crunch and crdao config
- added uusd support
- display tezosdomains name for recipient addresses
- integrated trading for STKR and wXTZ

### CODE REVIEW

File: src/contracts/KolibriToken/components/BalanceBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+                                scale={token.scale}
+                                precision={token.precision}
+                                round={token.round}
+                            />
+                            {pendingRewards !== '0.0' && (
+                                <div style={{ float: 'right' }}>
+                                    <span style={{ marginRight: '5px' }}>
+                                        {t('components.plenty.nouns.pending_rewards')}: {pendingRewards} kDAO
+                                    </span>
+                                    <SmallButton buttonTheme="primary" onClick={() => harvestTrigger()}>
```
Comment (dariuszsobkowicz) [inline src/contracts/KolibriToken/components/BalanceBanner.tsx:121]: you can use just fn reference to onClick if not passing any props

File: src/contracts/KolibriToken/components/HarvestModal.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    const onHarvest = async () => {
+        dispatch(harvestRewards(passPhrase));
+        onClose();
+    };
+
+    return (
+        <ModalWrapper open={open}>
+            {open ? (
+                <ModalContainer>
+                    <CloseIconWrapper onClick={() => onClose()} />
```
Comment (dariuszsobkowicz) [inline src/contracts/KolibriToken/components/HarvestModal.tsx:79]: fn reference only

File: src/contracts/Plenty/components/BalanceBanner.tsx [hunk 3]
```diff
@@ -70,17 +101,29 @@ function BalanceBanner(props: Props) {
...
+                                scale={token.scale}
+                                precision={token.precision}
+                                round={token.round}
+                            />
+                            {pendingRewards !== '0.0' && (
+                                <div style={{ float: 'right' }}>
+                                    <span style={{ marginRight: '5px' }}>
+                                        {t('components.plenty.nouns.pending_rewards')}: {pendingRewards}
+                                    </span>
+                                    <SmallButton buttonTheme="primary" onClick={() => harvestTrigger()}>
```
Comment (dariuszsobkowicz) [inline src/contracts/Plenty/components/BalanceBanner.tsx:121]: fn ref only

File: src/contracts/Plenty/components/HarvestModal.tsx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    const onHarvest = async () => {
+        dispatch(harvestRewards(passPhrase));
+        onClose();
+    };
+
+    return (
+        <ModalWrapper open={open}>
+            {open ? (
+                <ModalContainer>
+                    <CloseIconWrapper onClick={() => onClose()} />
```
Comment (dariuszsobkowicz) [inline src/contracts/Plenty/components/HarvestModal.tsx:79]: fn ref only

**Review by dariuszsobkowicz — APPROVED** [review]

File: src/contracts/KolibriToken/components/BalanceBanner.tsx (hunk 1 above)
Comment (anonymoussprocket) [reply src/contracts/KolibriToken/components/BalanceBanner.tsx:121]: the syntax re-writer might be doing this

**Review by vishakh — APPROVED** [review]

**Pull request merged**